build_scripts/api/helpers/image_builder.py

```
@@ -0,0 +1,271 @@
+                )
+
+                auto_tqdm = auto_shared.total_tqdm
+                shared.total_tqdm = pbar
+                pbar.reset(steps)
+                processed = process_txt2img(p)
+                p.close()
+                auto_shared.total_tqdm = auto_tqdm
+                output = processed
+            except:
+                print("No txt2img.")
+                self.use_txt2img = False
+        else:
+            with self.accelerator.autocast(), torch.inference_mode():
+                if seed is None or seed == '' or seed == -1:
+                    seed = int(random.randrange(0, 21474836147))
+
+                generator = torch.manual_seed(seed)
+                try:
+                    output = self.image_pipe(
+                        positive_prompts,
+                        num_inference_steps=steps,
+                        guidance_scale=scale,
+                        height=height,
+                        width=width,
+                        generator=generator,
+                        negative_prompt=negative_prompts).images
+                    self.exception_count = 0
+                except Exception as e:
+                    print(f"Exception generating images: {e}")
+                    traceback.print_exc()
+                    self.exception_count += 1
+                    if self.exception_count > 10:
+                        raise
+                    output = []
+                    pass
+
+        return output
+
+    def unload(self, is_ui):
+        # If we have an image pipe, delete it
+        if self.image_pipe is not None:
+            del self.image_pipe
+        if self.del_accelerator:
+            del self.accelerator
+        # If there was a model loaded already, reload it
+        if self.last_model is not None and not is_ui:
+            try:
+                from modules import sd_models
+                shared.sd_model = sd_models.load_model(self.last_model)
+            except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```